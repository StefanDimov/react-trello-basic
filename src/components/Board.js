import React from 'react'
import List from './List'
import { Navbar, Grid, Row, Col, Button } from 'react-bootstrap'

export default class Board extends React.Component {
    render() {
        const { title, lists, onCardClick, onCreateCardClick } = this.props

        const listElements = lists.map(list =>
            <Col key={list.title} sm={3}>
                <List title={list.title} cards={list.cards} onCardClick={onCardClick}>
                    <Button bsStyle="primary" block onClick={() => onCreateCardClick(list.id)}>Add Card</Button>
                </List>
            </Col>
        )

        return (
            <div>
                <Navbar fluid={true}>
                    <Navbar.Header>
                        <Navbar.Brand>{title}</Navbar.Brand>
                    </Navbar.Header>
                </Navbar>

                <Grid fluid={true}>
                    <Row>{listElements}</Row>
                </Grid>
            </div>
        )
    }
}

const { string, array, func } = React.PropTypes
Board.propTypes = {
    title: string.isRequired,
    lists: array.isRequired,
    onCardClick: func.isRequired,
    onCreateCardClick: func.isRequired
}
