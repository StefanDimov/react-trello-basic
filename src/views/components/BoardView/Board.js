import React from 'react'
import { Navbar, Nav, NavItem, Button, Glyphicon } from 'react-bootstrap'

import * as boardUtils from '../../../utils/boardUtils'
import CardList from './CardList'
import ListsWrapper from './ListsWrapper'
import AddNewListInput from './AddNewListInput'

/**
 * Component that visualizes a Board's lists and holds all the functionality regarding them.
 * @class comp/Board
 * @param {object} board The Board to be visualized
 * @param {function} onCardClick callback for when a card is clicked
 * @param {function} onCreateCard callback to which the data for a new card will be passed
 * @param {function} onCreateNewList callback to which the data for a new list will be passed
 * @param {function} onSaveBoard callback to which the data for updating a board will be passed
 */
export default class Board extends React.Component {
    render() { // eslint-disable-line
        const { board, onCardClick, onCreateCard, onCreateNewList, onSaveBoard } = this.props
        const { title, lists } = board

        const listElements = lists.map(list =>
            <CardList key={list.id} list={list} onCardClick={onCardClick}>
                <Button bsStyle="primary" block onClick={() => onCreateCard(list.id)}>Add Card</Button>
            </CardList>
        )

        return (
            <div className="my-navbar-top-padding">
                <Navbar fluid={true} fixedTop={true}>
                    <Navbar.Header>
                        <Navbar.Brand>{title}</Navbar.Brand>
                    </Navbar.Header>
                    <Nav pullRight={true}>
                        <NavItem onClick={() => { onSaveBoard(board) }}>
                            <Glyphicon glyph="floppy-disk" /> Save Board
                        </NavItem>
                        <NavItem href="/#">
                            <Glyphicon glyph="th-large" /> Boards
                        </NavItem>
                    </Nav>
                </Navbar>

                <ListsWrapper>
                    {listElements}
                    <AddNewListInput onCreateList={onCreateNewList} />
                </ListsWrapper>
            </div>
        )
    }
}

const { func } = React.PropTypes
Board.propTypes = {
    board: boardUtils.boardPropType.isRequired,
    onCardClick: func.isRequired,
    onCreateCard: func.isRequired,
    onCreateNewList: func.isRequired,
    onSaveBoard: func.isRequired
}
