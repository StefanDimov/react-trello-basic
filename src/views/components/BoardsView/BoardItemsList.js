import React from 'react'

import { Row, Col } from 'react-bootstrap'

import boardUtils from '../../../utils/boardUtils'
import BoardItem from './BoardItem'
import AddNewBoardItem from './AddNewBoardItem'

export default class BoardItemsList extends React.Component {
    render() {
        const { boards, onBoardClick, onCreateBoard } = this.props
        return (
            <Row>
                {boards.map((board) =>
                    <Col sm={3} key={board.id} onClick={() => onBoardClick(board)}>
                        <BoardItem board={board} />
                    </Col>
                )}
                <Col sm={3} key={'add-new-boards-item'}>
                    <AddNewBoardItem onCreateBoard={onCreateBoard} />
                </Col>
            </Row>
        )
    }
}

BoardItemsList.propTypes = {
    boards: React.PropTypes.arrayOf(boardUtils.boardPropType).isRequired,
    onBoardClick: React.PropTypes.func.isRequired,
    onCreateBoard: React.PropTypes.func.isRequired
}
