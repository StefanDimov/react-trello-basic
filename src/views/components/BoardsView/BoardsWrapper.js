import React from 'react'

import { Navbar, Grid, Row, Col } from 'react-bootstrap'

import boardUtils from '../../../utils/boardUtils'

import BoardItemsList from './BoardItemsList'

export default class BoardsWrapper extends React.Component {
    render() {
        const { boards, onBoardClick, onCreateBoard } = this.props

        return (
            <div className="my-navbar-top-padding">
                <Navbar fixedTop={true}>
                    <Navbar.Header>
                        <Navbar.Brand>Boards</Navbar.Brand>
                    </Navbar.Header>
                </Navbar>

                <Grid>
                    <Row>
                        <Col xs={12}>
                            <h2 className="h4">Choose a board...</h2>
                        </Col>
                    </Row>
                    <BoardItemsList boards={boards} onBoardClick={onBoardClick} onCreateBoard={onCreateBoard} />
                </Grid>
            </div>
        )
    }
}

BoardsWrapper.propTypes = {
    boards: React.PropTypes.arrayOf(boardUtils.boardPropType).isRequired,
    onBoardClick: React.PropTypes.func.isRequired,
    onCreateBoard: React.PropTypes.func.isRequired
}
