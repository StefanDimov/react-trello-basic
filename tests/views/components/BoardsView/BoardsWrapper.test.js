import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { getEmptyBoard } from '../../../_mocks/Board.mocks'

import BoardsWrapper from '../../../../src/views/components/BoardsView/BoardsWrapper'
import BoardItemsList from '../../../../src/views/components/BoardsView/BoardItemsList'

describe('BoardsWrapper', () => {

    const onBoardClick = jest.fn()
    const onCreateBoard = jest.fn()
    let board, anotherBoard, thirdBoard

    beforeEach(() => {
        board = getEmptyBoard(1)
        anotherBoard = getEmptyBoard(2)
        thirdBoard = getEmptyBoard(3)

        onBoardClick.mockClear()
        onCreateBoard.mockClear()
    })

    describe('rendering', () => {
        it('should render correctly with no boards', () => {
            const tree = renderer.create(
                <BoardsWrapper boards={[]} onBoardClick={onBoardClick} onCreateBoard={onCreateBoard} />
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })

        it('should render correctly with one board', () => {
            const tree = renderer.create(
                <BoardsWrapper boards={[board]} onBoardClick={onBoardClick} onCreateBoard={onCreateBoard} />
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })

        it('should render correctly with multiple boards', () => {
            const tree = renderer.create(
                <BoardsWrapper boards={[board, anotherBoard, thirdBoard]} onBoardClick={onBoardClick} onCreateBoard={onCreateBoard} />
            ).toJSON()

            expect(tree).toMatchSnapshot()
        })
    })

    describe('BoardItemsList', () => {
        it('should be passed correct params', () => {
            const wrapper = shallow(
                <BoardsWrapper boards={[board]} onBoardClick={onBoardClick} onCreateBoard={onCreateBoard} />
            )

            expect(wrapper.find(BoardItemsList).prop('onBoardClick')).toBe(onBoardClick)
            expect(wrapper.find(BoardItemsList).prop('onCreateBoard')).toBe(onCreateBoard)
        })
    })
})
