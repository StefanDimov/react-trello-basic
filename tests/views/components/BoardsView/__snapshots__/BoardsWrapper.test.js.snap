exports[`BoardsWrapper rendering should render correctly with multiple boards 1`] = `
<div
  className="my-navbar-top-padding">
  <nav
    className="navbar navbar-default navbar-fixed-top">
    <div
      className="container">
      <div
        className="navbar-header">
        <span
          className="navbar-brand">
          Boards
        </span>
      </div>
    </div>
  </nav>
  <div
    className="container">
    <div
      className="row">
      <div
        className="col-xs-12">
        <h2
          className="h4">
          Choose a board...
        </h2>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-sm-3"
        onClick={[Function]}>
        <div
          className="my-board-item">
          boardTitle1
        </div>
      </div>
      <div
        className="col-sm-3"
        onClick={[Function]}>
        <div
          className="my-board-item">
          boardTitle2
        </div>
      </div>
      <div
        className="col-sm-3"
        onClick={[Function]}>
        <div
          className="my-board-item">
          boardTitle3
        </div>
      </div>
      <div
        className="col-sm-3">
        <form
          className="my-add-new-board"
          onSubmit={[Function]}>
          <div
            className="form-group">
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Add new board..."
              type="text"
              value="" />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`BoardsWrapper rendering should render correctly with no boards 1`] = `
<div
  className="my-navbar-top-padding">
  <nav
    className="navbar navbar-default navbar-fixed-top">
    <div
      className="container">
      <div
        className="navbar-header">
        <span
          className="navbar-brand">
          Boards
        </span>
      </div>
    </div>
  </nav>
  <div
    className="container">
    <div
      className="row">
      <div
        className="col-xs-12">
        <h2
          className="h4">
          Choose a board...
        </h2>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-sm-3">
        <form
          className="my-add-new-board"
          onSubmit={[Function]}>
          <div
            className="form-group">
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Add new board..."
              type="text"
              value="" />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`BoardsWrapper rendering should render correctly with one board 1`] = `
<div
  className="my-navbar-top-padding">
  <nav
    className="navbar navbar-default navbar-fixed-top">
    <div
      className="container">
      <div
        className="navbar-header">
        <span
          className="navbar-brand">
          Boards
        </span>
      </div>
    </div>
  </nav>
  <div
    className="container">
    <div
      className="row">
      <div
        className="col-xs-12">
        <h2
          className="h4">
          Choose a board...
        </h2>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-sm-3"
        onClick={[Function]}>
        <div
          className="my-board-item">
          boardTitle1
        </div>
      </div>
      <div
        className="col-sm-3">
        <form
          className="my-add-new-board"
          onSubmit={[Function]}>
          <div
            className="form-group">
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Add new board..."
              type="text"
              value="" />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
